searx/webapp.py

```diff
@@ -354,6 +354,16 @@ def render(template_name, override_theme=None, **kwargs):
 
 @app.before_request
 def pre_request():
+    # request.request_data
+    if request.method == 'POST':
+        request_data = request.form
+    elif request.method == 'GET':
+        request_data = request.args
+    else:
+        request_data = {}
+
+    request.request_data = request_data
```

Comment:
It isn't necessary to select request data here. Below in this function all `GET` arguments are merged with `POST` arguments - see L366. From that point it is enough to use `request.form` it contains all the client arguments. Maybe the name `request.form` is confusing after the merge and `request.request_data` would be better, but merging arguments allows more flexible usage than selecting only `GET` or `POST` data exclusively.